src/sentry/data_export/tasks.py

```diff
@@ -354,12 +353,25 @@ def merge_export_blobs(data_export_id, **kwargs):
                 file.size = size
                 file.checksum = file_checksum.hexdigest()
                 file.save()
+
+            # This is in a separate atomic transaction because in prod, files exist
+            # outside of the primary database which means that the transaction to
+            # the primary database is idle the entire time the writes the the files
+            # database is happening. In the event the writes to the files database
+            # takes longer than the idle timeout, the connection to the primary
+            # database can timeout causing a failure.
+            with atomic_transaction(
+                using=(
+                    router.db_for_write(ExportedData),
+                    router.db_for_write(File),
+                )
+            ):
                 data_export.finalize_upload(file=file)
```

Comment:
This seems ok to me. One thought here is that we could potentially leave files dangling if `finalize_upload` fails. Would it make sense to put the indent back in place so that we only start the transaction to `ExportedData` once we need it, but if an exception is thrown we then rollback `File` and `FileBloxIndex`?